src/Chapter4.hs

```diff
@@ -629,7 +661,14 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM m1 m2 = do
+  a <- m1
+
+  if a then do
+    b <- m2
+
+    pure b
+  else pure a
```

Comment:
Great implementation! And a nice way of using the `do`-notation 💎 

If you want to practice with monads more, you can try implementing this function using the `>>=` operator directly, and it will be a nice one-liner 🙂 

Also, in the end, you don't need to return `pure a`, you can return `pure False`, because in the `else` branch you already know that `a` is `False`